### ISSUE TITLE
Different uint types not recognized in library functions.

### ISSUE THREAD

**Issue by androlo** [ethereum/solidity#379]
Maybe not an issue:

```
library TestLib {

    function test(uint v) constant returns (uint) {
        return v;
    }

    function test(uint16 v) constant returns (uint16) {
        return v;
    }
}

contract Test {

    using TestLib for *;

    function test(uint v) constant returns (uint) {
        return v.test();
    }

    function test(uint16 v) constant returns (uint16) {
        return v.test();
    }

}
```

Compiler error:

```
Untitled:21:16: Error: Member "test" not unique after argument-dependent lookup in uint16
        return v.test();
               ^----^
```

Same error when using `return TestLib.test(v);`, without `using`.

Overloading works fine and the library compiles. It also works fine if one type is `uint` and the other is `address` or some other type, but not if the other is `uintX`.

Online compiler: `Solidity version: 0.2.1-91a6b35f/.-Emscripten/clang/int`

**Comment by VoR0220** [comment]
The problem here is that you have two similarly named functions and size is not enough to distinguish a different function I'm afraid. Try naming the function differently :) appreciate the efforts though.

**Comment by chriseth** [comment]
`uint16` in implicitly convertible to both `uint16` and `uint`. Overload resolution only cares about whether the types are convertible at all. There is an item on the backlog to also take into account how "complicated" the conversion is or whether you need a conversion at all, so it might be fixed in 0.3.0.
